### ISSUE TITLE
HTML Middleware

### ISSUE THREAD

**Issue by apps-caraga** [mevdschee/php-crud-api#1011]
Not an issue, just a question.  Can be closed immediately if considered irrelevant. 😁 

I'm just been reading about htmx, how it supposedly simplifies things in webdev. I was just wondering if this API can also be used with htmx? Like, htmx needs a full or partial html document from the server, so would it be possible to have a middleware that produces that (html) instead of json or xml?

**Comment by mevdschee** [comment]
Awesome idea! It would need some examples I guess and a default representation (table for lists and dt/dd for records)?

**Comment by apps-caraga** [comment]
Yep, a default representation of tabular data would be useful. Although from what I read so far from its docs, almost any valid html tag/document can be returned, so that may complicate things up

**Comment by pat123456** [comment]
Hi, very interesting ! May it could be some default (table, td...) but if something else wanted : html template in a folder structure that reflex the url. Phpcrudapi will automatically check this folder and if find folders/html template parse them and return corresponding html. The template language could be inspirited by htmx syntax.
